pallets/eth-bridge/src/tx.rs

```diff
@@ -82,16 +83,17 @@ pub fn set_up_active_tx<T: Config>(req: SendRequestData) -> Result<(), Error<T>>
     let expiry = util::time_now::<T>() + EthTxLifetimeSecs::<T>::get();
     let extended_params = req.extend_params(expiry)?;
     let msg_hash = generate_msg_hash::<T>(&extended_params)?;
+    let new_sender = assign_sender()?;
 
     ActiveRequest::<T>::put(ActiveRequestData {
         request: Request::Send(req.clone()),
         confirmation: ActiveConfirmation { msg_hash, confirmations: BoundedVec::default() },
         tx_data: Some(ActiveEthTransaction {
-            function_name: req.function_name,
-            eth_tx_params: extended_params,
+            function_name: req.function_name.clone(),
+            eth_tx_params: extended_params.clone(),
             expiry,
             eth_tx_hash: H256::zero(),
-            sender: assign_sender()?,
+            sender: new_sender.clone(),
```

Comment:
why is this no longer inlined? I don't mind, but would like to know if there was a reason behind it

Reply:
So we can use it on the event.